pipenv/utils/shell.py

```diff
@@ -466,3 +461,37 @@ def shorten_path(location, bold=False):
     if bold:
         short[-1] = str(click.style(short[-1], bold=True))
     return os.sep.join(short)
+
+
+def isatty(stream):
+    try:
+        is_a_tty = stream.isatty()
+    except Exception:  # pragma: no cover
+        is_a_tty = False
+    return is_a_tty
+
+
+@contextmanager
+def cd(path) -> Iterator[None]:
```

Comment:
You can find `cd` in https://github.com/sarugaku/requirementslib/blob/400781f91d354f9f614f4d0b1e3c4187b3f3225c/src/requirementslib/fileutils.py#L24

Reply:
It is already included in reqlib 2.3.0 no need to define it here.